core/src/execution/providers/rpc.rs

```diff
@@ -263,10 +263,11 @@ impl<N: NetworkSpec, B: BlockProvider<N>, H: HistoricalBlockProvider<N>> BlockPr
         // 2. Try historical provider if available and not requesting latest
         if let Some(historical) = &self.historical_provider {
             if block_id != BlockNumberOrTag::Latest.into() {
-                if let Some(block) = historical
+                let historical_result = historical
                     .get_historical_block(block_id, full_tx, self)
-                    .await?
-                {
+                    .await;
```

Comment:
Let's leave it so that if the block fetch errors we return an error. Returning `None` has a special meaning of "we think there is no block here." and error should mean "we can't tell what the block is."